tests/test_save_fields.py

```diff
@@ -93,3 +93,15 @@ def test_correctly_handle_foreignkeys_id_field_in_update_fields():
 
     tmwfk.save(update_fields=['fkey'])
     assert tmwfk.get_dirty_fields(check_relationship=True) == {}
+
+
+@pytest.mark.django_db
+def test_save_deferred_field_with_update_fields():
+    TestModel.objects.create()
+
+    tm = TestModel.objects.defer('boolean').first()
+    tm.boolean = False
+
+    assert tm.get_dirty_fields() == {}
+
+    tm.save(update_fields=['boolean'])
```

Comment:
We should probably change the order of the lines, to assert that the deferred field is still ignored after the save:

```
tm = TestModel.objects.defer('boolean').first()
tm.save(update_fields=['boolean'])
tm.boolean = False
assert tm.get_dirty_fields() == {}
```
What do you think?